pkg/gadget-collection/gadgets/advise/seccomp/gadget.go

```diff
@@ -392,27 +392,28 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 			trace.Status.OperationError = fmt.Sprintf("Failed to start seccomp tracer: %s", err)
 			return
 		}
-
-		// 'trace' is owned by the controller and could be modified
-		// outside of the gadget control. Make a copy for the callback.
-		traceCopy := trace.DeepCopy()
-
-		// Subscribes to container termination events in order to
-		// generate a SeccompProfile when a container terminates. We
-		// don't need the list of existing containers, so the return
-		// value is ignored.
-		_ = t.helpers.Subscribe(
-			genPubSubKey(trace.ObjectMeta.Namespace+"/"+trace.ObjectMeta.Name),
-			*gadgets.ContainerSelectorFromContainerFilter(trace.Spec.Filter),
-			func(event containercollection.PubSubEvent) {
-				// Ignore container creation events.
-				if event.Type != containercollection.EventTypeRemoveContainer {
-					return
-				}
-				t.containerTerminated(traceCopy, event)
-			},
-		)
 	}
+
+	// 'trace' is owned by the controller and could be modified
+	// outside of the gadget control. Make a copy for the callback.
+	traceCopy := trace.DeepCopy()
+
+	// Subscribes to container termination events in order to
+	// generate a SeccompProfile when a container terminates. We
+	// don't need the list of existing containers, so the return
+	// value is ignored.
+	_ = t.helpers.Subscribe(
+		genPubSubKey(trace.ObjectMeta.Namespace+"/"+trace.ObjectMeta.Name),
+		*gadgets.ContainerSelectorFromContainerFilter(trace.Spec.Filter),
```

Comment:
With this change, the second instance of the seccomp gadget will completely overwrite the filter, which means that we will not generate the profile for `mypod1`. Is it what we want?

Reply:
Each trace (`gadgetv1alpha1.Trace`) object has a different filter that is used in `Subscribe()`. What this PR is making is that filters from another seccomp advisor instances are also taken into consideration, so when those pods terminate the profile is created.

Reply:
Aahhh ok, it is because each trace subscribes with a diff key, so there will be no overwritting at all. Got it!